CHANGELOG.md

```diff
@@ -1,5 +1,46 @@
+* Catalog Sync
+  * `kube-system` and `kube-public` namespaces are now synced from **unless** `-deny-k8s-namespace=kube-system -deny-k8s-namespace=kube-public` are passed to the `sync-catalog` command.
+  * Previously, multiple sync processes could be run in the same Kubernetes cluster with different source Kubernetes namespaces and the same `-consul-k8s-tag`. This is no longer possible.
+  The sync processes will now delete one-another's registrations. To continue running multiple sync processes, each process must be passed a different `-consul-k8s-tag` flag.
+  * Previously, catalog sync would delete services tagged with `-consul-k8s-tag` (defaults to `k8s`) that were registered out-of-band, i.e. not by the sync process itself. It would delete services regardless of which node they were registered on.
+  Now the sync process will only delete those services not registered by itself if they are on the `k8s-sync` node (the synthetic node created by the catalog sync process).
+
+IMPROVEMENTS:
+
+* Support [**Consul namespaces [Enterprise feature]**](https://www.consul.io/docs/enterprise/namespaces/index.html) in all consul-k8s components [[GH-197](https://github.com/hashicorp/consul-k8s/pull/197)]
+* Create allow and deny lists of k8s namespaces for catalog sync and Connect inject
+* Connect Inject
+  * Changes default Consul Docker image (`-consul-image`) to `consul:1.7.1`
+  * Changes default Envoy Docker image (`-envoy-image`) to `envoyproxy/envoy-alpine:v1.13.0`
+
+BUG FIXES:
+
+* Bootstrap ACLs: Allow users to update their Connect ACL binding rule definition on upgrade
+* Bootstrap ACLs: Fixes mesh gateway ACL policies to have the correct permissions
+* Sync: Fixes a hot loop bug when getting an error from Consul when retrieving service information
+
+DEPRECATIONS:
+* `connect-inject` flag `-create-inject-token` is deprecated in favor of new flag `-create-inject-auth-method`
+
+NOTES:
+
+* Bootstrap ACLs: Previously, ACL policies were not updated after creation. Now, if namespaces are enabled, they are updated every time the ACL bootstrapper is run so that any namespace config changes can be adjusted. This change is only an issue if you are updating ACL policies after creation.
+
+* Connect: Adds additional parsing of the upstream annotation to support namespaces. The format of the annotation becomes:
+
+    `service_name.optional_namespace:port:optional_datacenter`
+
+  The `service_name.namespace` is only parsed if namespaces are enabled. If they are not enabled and someone has added a `.namespace`, the upstream will not work correctly, as is the case when someone has put in an incorrect service name, port or datacenter. If namespaces are enabled and the `.namespace` is not defined, Consul will automatically fallback to assuming the service is in the same namespace as the service defining the upstream.
+
+* Connect: Due to a breaking config change in Envoy, if using Consul 1.7+, Envoy version 1.13+ is required. 
```

Comment:
I think it's a change in Consul that changed the minimum version of Envoy from 1.8 to 1.10. I'm also not sure it's required, since the docs mention that consul 1.7 is compatible with 1.10: https://www.consul.io/docs/connect/proxies/envoy.html#supported-versions.

Reply:
It's only compatible with an additional flag added to the `consul connect envoy` command that creates the bootstrap config. We put off adding that flag, so 1.10 is not currently compatible with our setup. Do you have a suggestion for how to phrase that nuance?

Also, didn't you mention that 1.13 was needed for namespaces?

Reply:
Ah, ok. You're right. I haven't tested all other versions. What I saw was that our default (1.9.1) didn't work with consul 1.7. It wasn't namespace specific. 

Also, this requirement applies to connect and mesh gateways.

Maybe something like:
```
Connect and Mesh Gateways: Consul 1.7+ now requires that we pass `-envoy-version` flag if using a version other than the default (1.13.0) so that it can generate correct bootstrap configuration. This is not yet supported in the Helm chart and consul-k8s, and as such, we require Envoy version 1.13.0. 
```

Reply:
Is this a breaking change in the helm chart? If someone is overwriting their envoy image, they'll see it break.

Reply:
Added this language into the breaking change section.